packages/taco/src/taco-config-validator.ts

```diff
@@ -0,0 +1,362 @@
+    if (!chainValidation.isValid) {
+      errors.push(...chainValidation.errors);
+    }
+
+    return { isValid: errors.length === 0, errors };
+  }
+
+  /**
+   * Synchronous chain compatibility validation
+   *
+   * Validates provider chain compatibility with domain requirements using
+   * synchronously available chain information.
+   *
+   * @param {TacoClientConfig} config - Configuration to validate
+   * @returns {ValidationResult} Validation result
+   * @private
+   */
+  private static validateChainCompatibility(
+    config: TacoClientConfig,
+  ): ValidationResult {
+    const errors: string[] = [];
+
+    // Get expected chain ID for domain
+    const expectedChainId = this.getExpectedChainId(config.domain);
+    if (!expectedChainId) {
+      errors.push(`Unsupported domain: ${config.domain}`);
+      return { isValid: false, errors };
+    }
+
+    // Check viem client chain compatibility
+    if ('viemClient' in config && config.viemClient) {
+      const viemClient = config.viemClient as PublicClient;
+      if (viemClient.chain && viemClient.chain.id !== expectedChainId) {
+        errors.push(
+          `Provider chain mismatch: viem client chain ID ${viemClient.chain.id} does not match domain '${config.domain}' (expected ${expectedChainId})`,
+        );
+      }
+      // Note: If viemClient.chain is undefined, we skip synchronous validation
+      // Full validation with validateFull() will perform the network check
+    }
+
+    // Check ethers provider chain compatibility
+    if ('ethersProvider' in config && config.ethersProvider) {
+      // Note: _network is not public API, but it's the only synchronous way to check
+      // However, if the property `_network` was not available, no error will be thrown.
+      const ethersProvider = config.ethersProvider as unknown as {
+        _network?: { chainId: number };
+      };
+      if (
+        ethersProvider._network &&
+        ethersProvider._network.chainId !== expectedChainId
+      ) {
+        errors.push(
+          `Provider chain mismatch: ethers provider chain ID ${ethersProvider._network.chainId} does not match domain '${config.domain}' (expected ${expectedChainId})`,
+        );
+      }
+    }
+
+    return { isValid: errors.length === 0, errors };
+  }
+
+  /**
+   * Full validation including async provider network checks
+   *
+   * Performs comprehensive validation including:
+   * - All fast validation checks
+   * - Async network calls to verify provider chain ID matches domain requirements
+   *
+   * Use this method when you need complete validation including network connectivity checks.
+   * For faster validation without network calls, use validateFast().
+   *
+   * @param {TacoClientConfig} config - Configuration to validate
+   * @returns {Promise<ValidationResult>} Promise resolving to validation result with isValid boolean and errors array
+   */
+  static async validateFull(
```

Comment:
IMHO, this should be the by-default validation function. So I find it more appropiate to simply call it `validate()`. If you want a non-network validation, then use the "special" method `fastValidate()`.

It sounds weird to me to have `fastValidate` and `validateFull` but not a `validate` function.

```suggestion
  static async validate(
```